### PR TITLE
python312Packages.bsddb3 : Patch specific test causing build failure for Athena-Nyx cybersecurity roles

### PR DESCRIPTION
**Pull request by AkechiShiro**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@D3vil0p3r 
---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AkechiShiro** [comment]
Should I close this @D3vil0p3r ?

**Comment by D3vil0p3r** [comment]
> Should I close this @D3vil0p3r ?

Why?

**Comment by AkechiShiro** [comment]
I thought it got fixed ? Or not yet ? I will then work toward fixing it here I guess, I'll just need to clean one commit

File: pkgs/development/python-modules/bsddb3/default.nix [hunk 1]
```diff
@@ -15,8 +15,10 @@ buildPythonPackage rec {
     sha256 = "70d05ec8dc568f42e70fc919a442e0daadc2a905a1cfb7ca77f549d49d6e7801";
   };
 
+
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/bsddb3/default.nix:18]: Keep formatting change in another commit ideally.

File: pkgs/development/python-modules/bsddb3/default.nix [hunk 2]
```diff
@@ -29,6 +31,13 @@ buildPythonPackage rec {
     export BERKELEYDB_DIR=${pkgs.db.dev};
   '';
 
+    postPatch = ''
+    substituteInPlace test3.py \
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/bsddb3/default.nix:35]: Indent once the line 35, look the rest of the file.

File: pkgs/development/python-modules/scrapy/default.nix [hunk 3]
```diff
@@ -82,6 +82,7 @@ buildPythonPackage rec {
   disabledTestPaths = [
     "tests/test_proxy_connect.py"
     "tests/test_utils_display.py"
+    #"tests/test_utils_python.py"
```
Comment (RaitoBezarius) [inline pkgs/development/python-modules/scrapy/default.nix:85]: If you comment them, you have to explain why you want to keep that comment. Alternatively, drop them!

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Your commit message would benefit from more clarity, please include a small cover letter in every of your PR bodies "description of changes" to give a desire for the reviewer to review your change.

File: pkgs/development/python-modules/scrapy/default.nix (hunk 3 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/scrapy/default.nix:85]: This was a leftover comment that's not needed anymore, will be removed.

**Comment by AkechiShiro** [comment]
@D3vil0p3r any insight/feedback for the commit message/description ? 

The main issue takes roots from a Python 3.12 breaking change : 

> PEP 632: Remove the distutils package. See the migration guide for advice replacing the APIs it provided. The third-party > Setuptools package continues to provide distutils, if you still require it in Python 3.12 and beyond.

Source : https://docs.python.org/3.12/whatsnew/3.12.html, https://peps.python.org/pep-0632/#migration-advice, https://peps.python.org/pep-0632/

# Fixing scrapy 

It was necessary that we ignore one specific test that was failing (`test_get_func_args`) we were not able to root cause the failure specific reason.

Another dependency test was also failing because of Python 3.12 distutils deprecation, 
and, we fixed the failing test in a dependency (bsddb3), with the following workaround : 
```nix
  postPatch = ''
    substituteInPlace test3.py \
      --replace-warn "from distutils.util import get_platform" "import platform" \
      --replace-warn "get_platform()" "f'{platform.system().lower()}-{platform.machine()}'" \
      --replace-warn "_TextTestResult" "TextTestResult"
  '';
```
Proper solution would be to : 
- Mark bsddb3 as broken because it is deprecated and replaced by a newer library upstream that is maintained, see : https://pypi.org/project/bsddb3/#description.
> This library is deprecated. If you are running Python 3 >= 3.6, please upgrade to berkeleydb library. Check details.
 

nixpkgs-review output : 
```nix
error: builder for '/nix/store/alzdy7cp83cgbfgwbndnykfyr1yg57z7-python3.12-pymilter-1.0.5.drv' failed with exit code 1;
       last 10 log lines:
       >     self.assertEquals(cache['temp@bar.com'],'testing')
       >     ^^^^^^^^^^^^^^^^^
       > AttributeError: 'AddrCacheTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
       >
       > ----------------------------------------------------------------------
       > Ran 27 tests in 0.186s
       >
       > FAILED (errors=6, expected failures=1)
       > Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       For full logs, run 'nix log /nix/store/alzdy7cp83cgbfgwbndnykfyr1yg57z7-python3.12-pymilter-1.0.5.drv'.
error: 1 dependencies of derivation '/nix/store/dgmmcx51ilwydpxky4bnba2kiv8h9ni8-review-shell.drv' failed to build
2 packages failed to build:
python312Packages.pymilter python312Packages.pymilter.dist

27 packages built:
exaile gramps gramps.dist python311Packages.bsddb3 python311Packages.bsddb3.dist python311Packages.pymilter python311Packages.pymilter.dist python311Packages.scrapy python311Packages.scrapy-deltafetch python311Packages.scrapy-deltafetch.dist python311Packages.scrapy-fake-useragent python311Packages.scrapy-fake-useragent.dist python311Packages.scrapy-splash python311Packages.scrapy-splash.dist python311Packages.scrapy.dist python312Packages.bsddb3 python312Packages.bsddb3.dist python312Packages.scrapy python312Packages.scrapy-deltafetch python312Packages.scrapy-deltafetch.dist python312Packages.scrapy-fake-useragent python312Packages.scrapy-fake-useragent.dist python312Packages.scrapy-splash python312Packages.scrapy-splash.dist python312Packages.scrapy.dist spf-engine spf-engine.dist
```

**Comment by AkechiShiro** [comment]
Additional testing shows Scrapy is missing the lxml dependency, I will fix it in the near future : 
```python3
>>> import scrapy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/47p02yz8rx40dj2fyr9f9rqjld86fwfi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/__init__.py", line 12, in <module>
    from scrapy.http import FormRequest, Request
  File "/nix/store/47p02yz8rx40dj2fyr9f9rqjld86fwfi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/http/__init__.py", line 10, in <module>
    from scrapy.http.request.form import FormRequest
  File "/nix/store/47p02yz8rx40dj2fyr9f9rqjld86fwfi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/http/request/form.py", line 11, in <module>
    from lxml.html import (
  File "/nix/store/p3gxjs6aqph6z20vj2pyblys0nkyg8g1-python3.12-lxml-5.1.0/lib/python3.12/site-packages/lxml/html/__init__.py", line 49, in <module>
    from .. import etree
ImportError: cannot import name 'etree' from 'lxml' (/nix/store/p3gxjs6aqph6z20vj2pyblys0nkyg8g1-python3.12-lxml-5.1.0/lib/python3.12/site-packages/lxml/__init__.py)
```

**Comment by AkechiShiro** [comment]
Official fix for the Scrapy test is here : https://github.com/scrapy/scrapy/commit/b1fe97dc6c8509d58b29c61cf7801eeee1b409a9 but hasn't landed in a release I believe yet.

**Comment by D3vil0p3r** [comment]
> Your commit message would benefit from more clarity, please include a small cover letter in every of your PR bodies "description of changes" to give a desire for the reviewer to review your change.

The commit message is clear imho. If more clarity is needed, the PR body could be good.

**Comment by D3vil0p3r** [comment]
@AkechiShiro I see this merged PR: https://github.com/NixOS/nixpkgs/pull/309729 Can you give a look if it already solved what you fixed here?

File: pkgs/development/python-modules/bsddb3/default.nix (hunk 1 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/bsddb3/default.nix:18]: Will keep this in mind for further PRs

File: pkgs/development/python-modules/bsddb3/default.nix (hunk 2 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/bsddb3/default.nix:35]: Is this good now ?

**Comment by AkechiShiro** [comment]
Thanks @D3vil0p3r, I will need to rebase and merge, will do the check soon.

**Comment by AkechiShiro** [comment]
Nixpkgs-review output : 
```nix
15 packages updated:
exaile gramps python3.11-bsddb3 python3.11-pymilter python3.11-scrapy python3.11-scrapy-deltafetch python3.11-scrapy-fake-useragent python3.11-scrapy-splash python3.12-bsddb3 python3.12-pymilter python3.12-scrapy python3.12-scrapy-deltafetch python3.12-scrapy-fake-useragent python3.12-scrapy-splash spf-engine

$ nix build --nix-path nixpkgs=/home/kenshin/.cache/nixpkgs-review/pr-308157-2/nixpkgs nixpkgs-overlays=/tmp/tmp8eanl8uh --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/kenshin/.cache/nixpkgs-review/pr-308157-2/build.nix
error: builder for '/nix/store/alzdy7cp83cgbfgwbndnykfyr1yg57z7-python3.12-pymilter-1.0.5.drv' failed with exit code 1;
       last 10 log lines:
       >     self.assertEquals(cache['temp@bar.com'],'testing')
       >     ^^^^^^^^^^^^^^^^^
       > AttributeError: 'AddrCacheTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
       >
       > ----------------------------------------------------------------------
       > Ran 27 tests in 0.380s
       >
       > FAILED (errors=6, expected failures=1)
       > Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       For full logs, run 'nix log /nix/store/alzdy7cp83cgbfgwbndnykfyr1yg57z7-python3.12-pymilter-1.0.5.drv'.
error: 1 dependencies of derivation '/nix/store/dgmmcx51ilwydpxky4bnba2kiv8h9ni8-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/308157

2 packages failed to build:
python312Packages.pymilter python312Packages.pymilter.dist

27 packages built:
exaile gramps gramps.dist python311Packages.bsddb3 python311Packages.bsddb3.dist python311Packages.pymilter python311Packages.pymilter.dist python311Packages.scrapy python311Packages.scrapy-deltafetch python311Packages.scrapy-deltafetch.dist python311Packages.scrapy-fake-useragent python311Packages.scrapy-fake-useragent.dist python311Packages.scrapy-splash python311Packages.scrapy-splash.dist python311Packages.scrapy.dist python312Packages.bsddb3 python312Packages.bsddb3.dist python312Packages.scrapy python312Packages.scrapy-deltafetch python312Packages.scrapy-deltafetch.dist python312Packages.scrapy-fake-useragent python312Packages.scrapy-fake-useragent.dist python312Packages.scrapy-splash python312Packages.scrapy-splash.dist python312Packages.scrapy.dist spf-engine spf-engine.dist
```

**Comment by AkechiShiro** [comment]
This PR is ready to be reviewed @D3vil0p3r @RaitoBezarius @fabaff

**Comment by AkechiShiro** [comment]
The wrapper seems to use Python3.11 but is should use Python 3.12, hmm not sure what change could fix this : 
```

[nix-shell:~/.cache/nixpkgs-review/pr-308157-3]$ python3.12
Python 3.12.3 (main, Apr  9 2024, 08:09:14) [GCC 13.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import scrapy
>>> 

[nix-shell:~/.cache/nixpkgs-review/pr-308157-3]$ scrapy
Traceback (most recent call last):
  File "/nix/store/ymc3plk4lz8qidcxldq8ags2j9c5r8f1-python3.11-scrapy-2.11.1/bin/.scrapy-wrapped", line 6, in <module>
    from scrapy.cmdline import execute
  File "/nix/store/9dxxyr1ajxgfk4xw5n7jmfhmipjycsmi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/__init__.py", line 12, in <module>
    from scrapy.http import FormRequest, Request
  File "/nix/store/9dxxyr1ajxgfk4xw5n7jmfhmipjycsmi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/http/__init__.py", line 10, in <module>
    from scrapy.http.request.form import FormRequest
  File "/nix/store/9dxxyr1ajxgfk4xw5n7jmfhmipjycsmi-python3.12-scrapy-2.11.1/lib/python3.12/site-packages/scrapy/http/request/form.py", line 11, in <module>
    from lxml.html import (
  File "/nix/store/p3gxjs6aqph6z20vj2pyblys0nkyg8g1-python3.12-lxml-5.1.0/lib/python3.12/site-packages/lxml/html/__init__.py", line 49, in <module>
    from .. import etree
ImportError: cannot import name 'etree' from 'lxml' (/nix/store/p3gxjs6aqph6z20vj2pyblys0nkyg8g1-python3.12-lxml-5.1.0/lib/python3.12/site-packages/lxml/__init__.py)
```

**Comment by AkechiShiro** [comment]
Do you approve of the current change @D3vil0p3r ?

File: pkgs/development/python-modules/bsddb3/default.nix [hunk 4]
```diff
@@ -1,9 +1,4 @@
-{ lib
-, buildPythonPackage
-, fetchPypi
-, pkgs
-, python
-}:
+{ lib, buildPythonPackage, fetchPypi, pkgs, python }:
```
Comment (fabaff) [inline pkgs/development/python-modules/bsddb3/default.nix:1]: Please keep it one entry per line.

File: pkgs/development/python-modules/bsddb3/default.nix [hunk 5]
```diff
@@ -29,10 +24,18 @@ buildPythonPackage rec {
     export BERKELEYDB_DIR=${pkgs.db.dev};
   '';
 
+  postPatch = ''
+    substituteInPlace test3.py \
+      --replace-warn "from distutils.util import get_platform" "import platform" \
+      --replace-warn "get_platform()" "f'{platform.system().lower()}-{platform.machine()}'" \
+      --replace-warn "_TextTestResult" "TextTestResult"
```
Comment (fabaff) [inline pkgs/development/python-modules/bsddb3/default.nix:31]: At first glance I would say that we should fail (`--replace-fail`) here.

File: pkgs/development/python-modules/bsddb3/default.nix [hunk 6]
```diff
@@ -29,10 +24,18 @@ buildPythonPackage rec {
...
+  '';
+
   meta = with lib; {
     description = "Python bindings for Oracle Berkeley DB";
     homepage = "https://www.jcea.es/programacion/pybsddb.htm";
-    license = with licenses; [ agpl3Only ]; # License changed from bsd3 to agpl3 since 6.x
+    license = with licenses;
+      [ agpl3Only ]; # License changed from bsd3 to agpl3 since 6.x
```
Comment (fabaff) [inline pkgs/development/python-modules/bsddb3/default.nix:38]: The comment is not really useful anymore. 

```suggestion
    license = licenses.agpl3Only;
```

File: pkgs/development/python-modules/scrapy/default.nix [hunk 7]
```diff
@@ -49,15 +22,13 @@ buildPythonPackage rec {
     # fix test_get_func_args. remove on next update
     (fetchpatch {
       name = "test_get_func_args.patch";
-      url = "https://github.com/scrapy/scrapy/commit/b1fe97dc6c8509d58b29c61cf7801eeee1b409a9.patch";
+      url =
+        "https://github.com/scrapy/scrapy/commit/b1fe97dc6c8509d58b29c61cf7801eeee1b409a9.patch";
```
Comment (fabaff) [inline pkgs/development/python-modules/scrapy/default.nix:26]: Please revert this. `nixfmt` would probably to it in the future.

```suggestion
      url = "https://github.com/scrapy/scrapy/commit/b1fe97dc6c8509d58b29c61cf7801eeee1b409a9.patch";
```

File: pkgs/development/python-modules/scrapy/default.nix [hunk 8]
```diff
@@ -49,15 +22,13 @@ buildPythonPackage rec {
...
       hash = "sha256-POlmsuW4SD9baKwZieKfmlp2vtdlb7aKQ62VOmNXsr0=";
     })
   ];
 
-  nativeBuildInputs = [
-    installShellFiles
-    setuptools
-  ];
+  nativeBuildInputs = [ installShellFiles setuptools ];
```
Comment (fabaff) [inline pkgs/development/python-modules/scrapy/default.nix:31]:
```suggestion
  build-system = [ setuptools ];

  nativeBuildInputs = [ installShellFiles ];
```

File: pkgs/development/python-modules/scrapy/default.nix [hunk 9]
```diff
@@ -1,35 +1,8 @@
...
-, tldextract
-, twisted
-, w3lib
-, zope-interface
-}:
+{ lib, stdenv, botocore, buildPythonPackage, cryptography, cssselect, fetchPypi
+, glibcLocales, installShellFiles, itemadapter, itemloaders, jmespath, lxml
+, fetchpatch, packaging, parsel, pexpect, protego, pydispatcher, pyopenssl
+, pytestCheckHook, pythonOlder, queuelib, service-identity, setuptools, sybil
+, testfixtures, tldextract, twisted, w3lib, zope-interface }:
```
Comment (fabaff) [inline pkgs/development/python-modules/scrapy/default.nix:5]: Please keep it one entry per line. Multiple entries per per line can't be sorted in a easy way and reviewing is harder.

**Comment by D3vil0p3r** [comment]
bsddb3 seems to be fixed here: https://github.com/NixOS/nixpkgs/issues/308232

**Comment by D3vil0p3r** [comment]
@AkechiShiro @fabaff please give a look to the last applied commit in scrapy. It seems to work now. Probably this PR can be closed.

**Comment by AkechiShiro** [comment]
@D3vil0p3r I believe this PR is better as we fixed the test and did not disable it.

File: pkgs/development/python-modules/scrapy/default.nix (hunk 9 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/scrapy/default.nix:5]: Thanks, we'll keep this in mind, nixfmt just did what are its current default at the moment.

**Comment by D3vil0p3r** [comment]
> @D3vil0p3r I believe this PR is better as we fixed the test and did not disable it.

Can you please apply the suggestions of @fabaff ?

File: pkgs/development/python-modules/bsddb3/default.nix (hunk 6 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/bsddb3/default.nix]: Done

File: pkgs/development/python-modules/scrapy/default.nix (hunk 7 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/scrapy/default.nix]: Fixed

File: pkgs/development/python-modules/scrapy/default.nix (hunk 8 above)
Comment (AkechiShiro) [reply pkgs/development/python-modules/scrapy/default.nix]: Done

**Comment by AkechiShiro** [comment]
Was working on it @D3vil0p3r, I think everything is good now.

If you have time for a review @fabaff, feel free to let me know if it's now good ?

**Review by D3vil0p3r — APPROVED** [review]

**Comment by AkechiShiro** [comment]
Nixpkgs-review output (LGTM) : 
```nix
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/kenshin/.cache/nixpkgs-review/pr-308157-7/nixpkgs nixpkgs-overlays=/tmp/tmpq91akokq -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
9 packages updated:
exaile gramps python3.11-bsddb3 python3.11-pymilter python3.11-scrapy-deltafetch python3.12-bsddb3 python3.12-pymilter python3.12-scrapy-deltafetch spf-engine

$ nix build --nix-path nixpkgs=/home/kenshin/.cache/nixpkgs-review/pr-308157-7/nixpkgs nixpkgs-overlays=/tmp/tmpq91akokq --extra-experimental-features nix-command no-url-literals --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed -f /home/kenshin/.cache/nixpkgs-review/pr-308157-7/build.nix
error: builder for '/nix/store/4b0yac479mdwca75ggwa4d276r1xhzwc-python3.12-pymilter-1.0.5.drv' failed with exit code 1;
       last 10 log lines:
       >     self.assertEquals(cache['temp@bar.com'],'testing')
       >     ^^^^^^^^^^^^^^^^^
       > AttributeError: 'AddrCacheTestCase' object has no attribute 'assertEquals'. Did you mean: 'assertEqual'?
       >
       > ----------------------------------------------------------------------
       > Ran 27 tests in 0.357s
       >
       > FAILED (errors=6, expected failures=1)
       > Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       > error: Test failed: <unittest.runner.TextTestResult run=27 errors=6 failures=0>
       For full logs, run 'nix log /nix/store/4b0yac479mdwca75ggwa4d276r1xhzwc-python3.12-pymilter-1.0.5.drv'.
error: 1 dependencies of derivation '/nix/store/x91p0xxgylixarbp6vdqdjqavg8fpqd3-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/308157

2 packages failed to build:
python312Packages.pymilter python312Packages.pymilter.dist

15 packages built:
exaile gramps gramps.dist python311Packages.bsddb3 python311Packages.bsddb3.dist python311Packages.pymilter python311Packages.pymilter.dist python311Packages.scrapy-deltafetch python311Packages.scrapy-deltafetch.dist python312Packages.bsddb3 python312Packages.bsddb3.dist python312Packages.scrapy-deltafetch python312Packages.scrapy-deltafetch.dist spf-engine spf-engine.dist
```

**Comment by AkechiShiro** [comment]
I opened another PR to fix pymilter's failing test here #311086

**Comment by AkechiShiro** [comment]
This PR will be closed as and superseeded by #311198
